src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -279,89 +287,48 @@ HRESULT STDMETHODCALLTYPE CorProfiler::AssemblyLoadFinished(AssemblyID assembly_
     const auto assembly_info = GetAssemblyInfo(this->info_, assembly_id);
     if (!assembly_info.IsValid())
     {
-        Logger::Debug("AssemblyLoadFinished: ", assembly_id, " ", hr_status);
         return S_OK;
     }
 
-    const auto is_instrumentation_assembly = assembly_info.name == WStr("OpenTelemetry.AutoInstrumentation");
-
-    if (is_instrumentation_assembly)
+    const auto is_instrumentation_assembly = assembly_info.name == managed_profiler_name;
+    if (!is_instrumentation_assembly)
     {
-        if (Logger::IsDebugEnabled())
-        {
-            Logger::Debug("AssemblyLoadFinished: ", assembly_id, " ", hr_status);
-        }
+        return S_OK;
+    }
 
-        ComPtr<IUnknown> metadata_interfaces;
-        auto hr = this->info_->GetModuleMetaData(assembly_info.manifest_module_id, ofRead | ofWrite,
-                                                 IID_IMetaDataImport2, metadata_interfaces.GetAddressOf());
+    if (Logger::IsDebugEnabled())
+    {
+        Logger::Debug("AssemblyLoadFinished: Bytecode Instrumentation Assembly: ", GetBytecodeInstrumentationAssembly());
+    }
 
-        if (FAILED(hr))
-        {
-            Logger::Warn("AssemblyLoadFinished failed to get metadata interface for module id ",
-                         assembly_info.manifest_module_id, " from assembly ", assembly_info.name);
-            return S_OK;
-        }
+    ComPtr<IUnknown> metadata_interfaces;
+    auto hr = this->info_->GetModuleMetaData(assembly_info.manifest_module_id, ofRead | ofWrite,
+                                                IID_IMetaDataImport2, metadata_interfaces.GetAddressOf());
+    if (FAILED(hr))
+    {
+        Logger::Warn("AssemblyLoadFinished failed to get metadata interface for module id ",
+                        assembly_info.manifest_module_id, " from assembly ", assembly_info.name);
+        return S_OK;
+    }
 
-        // Get the IMetaDataAssemblyImport interface to get metadata from the managed assembly
-        const auto assembly_import = metadata_interfaces.As<IMetaDataAssemblyImport>(IID_IMetaDataAssemblyImport);
-        const auto assembly_metadata = GetAssemblyImportMetadata(assembly_import);
+    // Get the IMetaDataAssemblyImport interface to get metadata from the managed assembly
+    const auto assembly_import = metadata_interfaces.As<IMetaDataAssemblyImport>(IID_IMetaDataAssemblyImport);
+    const auto assembly_metadata = GetAssemblyImportMetadata(assembly_import);
 
-        // used multiple times for logging
-        const auto assembly_version = assembly_metadata.version.str();
+    managed_profiler_loaded_app_domains.insert(assembly_info.app_domain_id);
```

Comment:
It looks like many of the version checks were removed that used to occur before this line of code was executed. Were those checks not necessary?

Reply:
Upstream shipped that same assembly as a NuGet package for manual instrumentation. In that case it was possible that the bytecode instrumentation was expecting something different of the assembly so it had to enforce which was loaded.

Eventually we may want to ship that assembly so people can write their instrumentations but I would rather design it when we get there.